openslides_votecollector/views.py

```diff
@@ -463,15 +557,23 @@ def post(self, request, poll_id, keypad_id):
             return HttpResponse(_('Vote rejected'))
 
         if vc.voting_mode == 'MotionPoll':
-            try:
-                conn = MotionPollKeypadConnection.objects.get(poll=poll, keypad=keypad)
-            except MotionPollKeypadConnection.DoesNotExist:
-                conn = MotionPollKeypadConnection()
-                conn.poll = poll
-                conn.keypad = keypad
-            conn.serial_number = request.POST.get('sn')
-            conn.value = value
-            conn.save()
+            is_valid_keypad = 1
+            if Ballot:
+                ballot = Ballot(poll)
+                is_valid_keypad = ballot.register_vote(keypad_id, value) > 0
+            if is_valid_keypad:
+                try:
+                    conn = MotionPollKeypadConnection.objects.get(poll=poll, keypad=keypad)
+                except MotionPollKeypadConnection.DoesNotExist:
+                    conn = MotionPollKeypadConnection()
+                    conn.poll = poll
+                    conn.keypad = keypad
+                conn.serial_number = request.POST.get('sn')
+                conn.value = value
+                conn.save()
+            else:
+                return HttpResponse(_('Vote rejected'))
```

Comment:
Please try not to return anything deep inside the function. The code would be much more readable.